## TikTok's Claims Classification Project

**Inspection and Analysis**

$Task:$ Inspect and understand the dataset.

In [1]:
# Import the packages
import os
import pandas as pd
import numpy as np

In [2]:
# Load the dataset
get_cwd = os.getcwd()
data = pd.read_csv(get_cwd + '/tiktok_dataset.csv')

In [3]:
# Examine the first ten rows of the dataframe
data.head()

,#,claim_status,video_id,video_duration_sec,video_transcription_text,verified_status,author_ban_status,video_view_count,video_like_count,video_share_count,video_download_count,video_comment_count
0,1,claim,7017666017,59,someone shared with me that drone deliveries a...,not verified,under review,343296.0,19425.0,241.0,1.0,0.0
1,2,claim,4014381136,32,someone shared with me that there are more mic...,not verified,active,140877.0,77355.0,19034.0,1161.0,684.0
2,3,claim,9859838091,31,someone shared with me that american industria...,not verified,active,902185.0,97690.0,2858.0,833.0,329.0
3,4,claim,1866847991,25,someone shared with me that the metro of st. p...,not verified,active,437506.0,239954.0,34812.0,1234.0,584.0
4,5,claim,7105231098,19,someone shared with me that the number of busi...,not verified,active,56167.0,34987.0,4110.0,547.0,152.0


In [4]:
# Get summary info
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19382 entries, 0 to 19381
Data columns (total 12 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   #                         19382 non-null  int64  
 1   claim_status              19084 non-null  object 
 2   video_id                  19382 non-null  int64  
 3   video_duration_sec        19382 non-null  int64  
 4   video_transcription_text  19084 non-null  object 
 5   verified_status           19382 non-null  object 
 6   author_ban_status         19382 non-null  object 
 7   video_view_count          19084 non-null  float64
 8   video_like_count          19084 non-null  float64
 9   video_share_count         19084 non-null  float64
 10  video_download_count      19084 non-null  float64
 11  video_comment_count       19084 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 1.8+ MB


In [5]:
# Get summary statistics
data.describe()

,#,video_id,video_duration_sec,video_view_count,video_like_count,video_share_count,video_download_count,video_comment_count
count,19382.000000,1.938200e+04,19382.000000,19084.000000,19084.000000,19084.000000,19084.000000,19084.000000
mean,9691.500000,5.627454e+09,32.421732,254708.558688,84304.636030,16735.248323,1049.429627,349.312146
std,5595.245794,2.536440e+09,16.229967,322893.280814,133420.546814,32036.174350,2004.299894,799.638865
min,1.000000,1.234959e+09,5.000000,20.000000,0.000000,0.000000,0.000000,0.000000
25%,4846.250000,3.430417e+09,18.000000,4942.500000,810.750000,115.000000,7.000000,1.000000
50%,9691.500000,5.618664e+09,32.000000,9954.500000,3403.500000,717.000000,46.000000,9.000000
75%,14536.750000,7.843960e+09,47.000000,504327.000000,125020.000000,18222.000000,1156.250000,292.000000
max,19382.000000,9.999873e+09,60.000000,999817.000000,657830.000000,256130.000000,14994.000000,9599.000000


1. The dataframe contains a collection of categorical, text, and numerical data. Each row represents a distinct TikTok video that presents either a claim or an opinion and the accompanying metadata about that video.

2. The dataframe contains five float64s, three int64s, and four objects. There are 19,382 observations, but some of the variables are missing values, including claim status, the video transcripton, and all of the count variables.

3. Many of the count variables seem to have outliers at the high end of the distribution. They have very large standard deviations and maximum values that are very high compared to their quartile values.

In [6]:
# The different values for claim status and how many of each are in the data?
total = data.count()[0]
claim = data['claim_status'].value_counts()[0]
opinion = data['claim_status'].value_counts()[1]
print("'claim': ", claim)
print("'opinion': ", opinion)
print("Total claim_status: ", claim + opinion)
print('Null Values: ', total - (claim + opinion))

'claim':  9608
'opinion':  9476
Total claim_status:  19084
Null Values:  298


C:\Users\222817\AppData\Local\Temp\ipykernel_1736\225789752.py:2: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  total = data.count()[0]
C:\Users\222817\AppData\Local\Temp\ipykernel_1736\225789752.py:3: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  claim = data['claim_status'].value_counts()[0]
C:\Users\222817\AppData\Local\Temp\ipykernel_1736\225789752.py:4: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  opinion = data['claim_status'

In [7]:
data['claim_status'].value_counts(normalize=True)*100

claim_status
claim      50.345839
opinion    49.654161
Name: proportion, dtype: float64

The counts of each claim status are quite balanced.

In [8]:
# The average view count of videos with "claim" status?
claim_data = data[data['claim_status'] == 'claim']
print("Mean: ", round(claim_data['video_view_count'].mean(), 2))
print("Median: ", claim_data['video_view_count'].median())

Mean:  501029.45
Median:  501555.0


In [9]:
# The average view count of videos with "opinion" status?
opinion_data = data[data['claim_status'] == 'opinion']
print("Mean: ", round(opinion_data['video_view_count'].mean(), 2))
print("Median: ", opinion_data['video_view_count'].median())

Mean:  4956.43
Median:  4953.0


The mean and the median within each claim category are close to one another, but there is a vast discrepancy between view counts for videos labeled as claims and videos labeled as opinions.

In [10]:
# Get counts for each group combination of claim status and author ban status
data.groupby(['claim_status', 'author_ban_status']).count()[['#']]

#
claim_status author_ban_status      
claim        active             6566
             banned             1439
             under review       1603
opinion      active             8817
             banned              196
             under review        463

There are many more claim videos with banned authors than there are opinion videos with banned authors. This could mean a number of things, including the possibilities that:

- Claim videos are more strictly policed than opinion videos.
- Authors must comply with a stricter set of rules if they post a claim than if they post an opinion.

In [11]:
# What's the median video share count of each author active status?
print("Active Status Count: ", round(data[data['author_ban_status'] == 'active']['video_view_count'].mean(), 2))

Active Status Count:  215927.04


In [12]:
# What's the median video share count of each author ban status?
print("Banned Status Count: ", round(data[data['author_ban_status'] == 'banned']['video_view_count'].mean(), 2))

Banned Status Count:  445845.44


In [13]:
# What's the median video share count of each author under review status?
print("Under review Status Count: ", round(data[data['author_ban_status'] == 'under review']['video_view_count'].mean(), 2))

Under review Status Count:  392204.84


Banned authors have a median share count that's 33 times the median share count of active authors! Explore this in more depth.

In [14]:
# Get the mean, median, count for video_view_count, video_like_count and video_share_count for each author active status
data.groupby('author_ban_status').agg(
    {'video_view_count': ['count', 'mean', 'median'],
     'video_like_count': ['count', 'mean', 'median'],
     'video_share_count': ['count', 'mean', 'median']})

video_view_count                          video_like_count  \
                             count           mean    median            count   
author_ban_status                                                              
active                       15383  215927.039524    8616.0            15383   
banned                        1635  445845.439144  448201.0             1635   
under review                  2066  392204.836399  365245.5             2066   

                                           video_share_count                \
                            mean    median             count          mean   
author_ban_status                                                            
active              71036.533836    2222.0             15383  14111.466164   
banned             153017.236697  105573.0              1635  29998.942508   
under review       128718.050339   71204.5              2066  25774.696999   

                            
                    median  
author_ban_status           
active               437.0  
banned             14468.0  
under review        9444.0

* Banned authors and those under review get far more views, likes, and shares than active authors.
* In most groups, the mean is much greater than the median, which indicates that there are some videos with very high engagement counts.

In [15]:
# Create a likes_per_view column
data['likes_per_view'] = data['video_like_count']/data['video_view_count']
# Create a comments_per_view column
data['comments_per_view'] = data['video_comment_count']/data['video_view_count']
# Create a shares_per_view column
data['shares_per_view'] = data['video_share_count']/data['video_view_count']

In [16]:
data.head()

,#,claim_status,video_id,video_duration_sec,video_transcription_text,verified_status,author_ban_status,video_view_count,video_like_count,video_share_count,video_download_count,video_comment_count,likes_per_view,comments_per_view,shares_per_view
0,1,claim,7017666017,59,someone shared with me that drone deliveries a...,not verified,under review,343296.0,19425.0,241.0,1.0,0.0,0.056584,0.000000,0.000702
1,2,claim,4014381136,32,someone shared with me that there are more mic...,not verified,active,140877.0,77355.0,19034.0,1161.0,684.0,0.549096,0.004855,0.135111
2,3,claim,9859838091,31,someone shared with me that american industria...,not verified,active,902185.0,97690.0,2858.0,833.0,329.0,0.108282,0.000365,0.003168
3,4,claim,1866847991,25,someone shared with me that the metro of st. p...,not verified,active,437506.0,239954.0,34812.0,1234.0,584.0,0.548459,0.001335,0.079569
4,5,claim,7105231098,19,someone shared with me that the number of busi...,not verified,active,56167.0,34987.0,4110.0,547.0,152.0,0.622910,0.002706,0.073175


In [17]:
# Get the mean, median, count for likes_per_view, comments_per_view and shares_per_view for each claim_status and author_ban_status
data.groupby(['claim_status', 'author_ban_status']).agg(
    {'likes_per_view': ['mean', 'median', 'count'],
     'comments_per_view': ['mean', 'median', 'count'],
     'shares_per_view': ['mean', 'median', 'count'],})

likes_per_view                  \
                                         mean    median count   
claim_status author_ban_status                                  
claim        active                  0.329542  0.326538  6566   
             banned                  0.345071  0.358909  1439   
             under review            0.327997  0.320867  1603   
opinion      active                  0.219744  0.218330  8817   
             banned                  0.206868  0.198483   196   
             under review            0.226394  0.228051   463   

                               comments_per_view                  \
                                            mean    median count   
claim_status author_ban_status                                     
claim        active                     0.001393  0.000776  6566   
             banned                     0.001377  0.000746  1439   
             under review               0.001367  0.000789  1603   
opinion      active                     0.000517  0.000252  8817   
             banned                     0.000434  0.000193   196   
             under review               0.000536  0.000293   463   

                               shares_per_view                  
                                          mean    median count  
claim_status author_ban_status                                  
claim        active                   0.065456  0.049279  6566  
             banned                   0.067893  0.051606  1439  
             under review             0.065733  0.049967  1603  
opinion      active                   0.043729  0.032405  8817  
             banned                   0.040531  0.030728   196  
             under review             0.044472  0.035027   463

We know that videos by banned authors and those under review tend to get far more views, likes, and shares than videos by non-banned authors. However, when a video does get viewed, its engagement rate is less related to author ban status and more related to its claim status.

Also, we know that claim videos have a higher view rate than opinion videos, but this tells us that claim videos also have a higher rate of likes on average, so they are more favorably received as well. Furthermore, they receive more engagement via comments and shares than opinion videos.

Note that for claim videos, banned authors have slightly higher likes/view and shares/view rates than active authors or those under review. However, for opinion videos, active authors and those under review both get higher engagement rates than banned authors in all categories.

* Of the 19,382 samples in this dataset, just under 50% are claims&mdash;9,608 of them.  
* Engagement level is strongly correlated with claim status. This should be a focus of further inquiry.
* Videos with banned authors have significantly higher engagement than videos with active authors. Videos with authors under review fall between these two categories in terms of engagement levels.